plugins/lenovo-firmware/fu-plugin-lenovo-firmware.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2021 Mario Limonciello <mario.limonciello@amd.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_BETTER_THAN, "bios");
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	g_autofree gchar *sysfsfwdir = NULL;
+	g_autofree gchar *thinklmidir = NULL;
+
+	/* already exists */
+	sysfsfwdir = fu_common_get_path (FU_PATH_KIND_SYSFSDIR_FW_ATTRIB);
+	thinklmidir = g_build_filename (sysfsfwdir, "thinklmi", NULL);
```

Comment:
do we have to "watch" this? i.e. can you change it at runtime?

Reply:
You can certainly _change_ it in `thinklmi` runtime, but whether it's effective before the next boot, that's a question for @mrhpearson to find out the answer.

I notice that thinklmi is missing a `reboot_required` attribute, which I think if things aren't effective until the next boot the driver probably should add if posisble.

Reply:
Hi,
A reboot is required I'm afraid.

We don't have the reboot_required attribute yet as the support isn't provided by the BIOS. At the moment everything needs a reboot (that's certainly been my experience in testing).

I thought about adding it to the driver - but was worried that SW shouldn't really be doing something that the BIOS should provide. I'm hoping the BIOS team add it in the future. If you think that's incorrect and we should add the attribute and set it whenever something is changed happy to do that.

Reply:
@mrhpearson just to make sure you didn't miss this - can you please check explicitly if changing it requires a reboot to be effective or it's immediate?  If it's reboot required, I think you need to also populate
`/sys/class/firmware-attributes/*/attributes/pending_reboot` on any attribute change and this plugin should look for it.

Reply:
>A reboot is required I'm afraid.

No problem, that was actually my expectation too.

>We don't have the reboot_required attribute yet as the support isn't provided by the BIOS. At the moment everything needs a reboot (that's certainly been my experience in testing).
> I thought about adding it to the driver - but was worried that SW shouldn't really be doing something that the BIOS should provide. I'm hoping the BIOS team add it in the future. If you think that's incorrect and we should add the attribute and set it whenever something is changed happy to do that.

Yes - I think it's important.  If you look at this from a userspace software perspective, userspace software should be able to say something along the lines of, this is what is set for the next boot, and options might not apply yet.
I think that fwupd should look `reboot_pending` and prevent updates until the value goes from 1->0 from the next boot to be safe.

FWIW Dell BIOS doesn't provide it either, and dell-wmi-sysman just caches it based on there being a change.  I think this is perfectly fine to do in `think-lmi` as well.

Reply:
Also, IIRC `dell-wmi-sysman` will emit a `CHANGE` `uevent` that userspace can monitor for.   to @hughsie's point, basically if that exists then we don't need to poll the file or anything.  We monitor for that change uevent and if it comes through then we run a callback to add inhibits.

Reply:
Ack - thanks. I'll look at adding that then. Thanks for the notes :)
I'll try and get that done this week but I've got a couple of half day events coming up, end of school year stuff and a vaccine jab (which last time knocked me out for a day) so it might slide into next week.